### PR TITLE
Cgranade/0.6

### PR DESCRIPTION
**Pull request by cgranade**

### CODE REVIEW

**Comment by cgranade** [comment]
This is currently generating a lot of invalid cross-references due to depending on #405 for the new API docs. I'm confused why we're not getting the right cross-references into the .NET API Browser, though. @cpalmer2020 @MikeDodaro: Might be good to ask if we need to modify `docfx.json` for that?

**Comment by MikeDodaro** [comment]
We asked Shir Goldberg, who helped with doc feedback option, about this problem.

File: articles/libraries/standard/convert.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# Type Conversions #
+
+Q# is a **strongly-typed** language.
+In particular, Q# does not implicitly cast between distinct types, such that `1 + 2.0` is not a valid Q# expression.
+Rather, Q# provides a variety of type conversion functions for constructing new values of a given type.
+
+For example, <xref:microsoft.quantum.core.length> has an output type of `Int`, such that its output must first be converted to a `Double` before it can be used as a part of a floating-point expression.
+This can be done using the <xref:microsoft.quantum.convert.intasdouble> function:
+
+```Q#
```
Comment (alan-geller) [inline articles/libraries/standard/convert.md]:
```suggestion
```qsharp
```

File: articles/libraries/standard/convert.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+open Microsoft.Quantum.Convert as Convert;
+
+function HalfLength<'T>(arr : 'T[]) : Double {
+    return Convert.IntAsDouble(Length(arr)) / 2.0;
+}
+```
+
+The <xref:microsoft.quantum.convert> namespace provides common type conversion functions for working with the basic built-in types, such as `Int`, `Double`, `BigInt`, `Result`, and `Bool`:
+
+```Q#
```
Comment (alan-geller) [inline articles/libraries/standard/convert.md]:
```suggestion
```qsharp
```

File: articles/libraries/standard/diagnostics.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+uid: microsoft.quantum.libraries.diagnostics
+ms.author: chgranad@microsoft.com 
+ms.topic: article
+---
+
+# Diagnostics #
+
+As with classical development, it is important to be able to diagnose mistakes and errors in quantum programs.
+The Q# standard libraries provide a variety of different ways to ensure the correctness of quantum programs, as detailed in <xref:microsoft.quantum.techniques.testing-and-debugging>.
+Largely speaking, this support comes in the form of functions and operations that either instruct target machines to provide additional diagnostic information to host programs, or that enforce the correctness of conditions and invariants expressed by quantum programmers.
```
Comment (alan-geller) [inline articles/libraries/standard/diagnostics.md]:
```suggestion
Largely speaking, this support comes in the form of functions and operations that either instruct the target machine to provide additional diagnostic information to the host program or developer, or enforce the correctness of conditions and invariants expressed by the function or operation call.
```

File: articles/libraries/standard/diagnostics.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+By default, this writes the string to the console.
+Used together with interpolated strings, <xref:microsoft.quantum.intrinsic.message> makes it easy to report diagnostic information about classical values:
+
+```Q#
+let angle = Microsoft.Quantum.Math.PI() * 2.0 / 3.0;
+Message($"About to rotate by an angle of {angle}...");
+```
+
+> [!NOTE]
+> `Message` has signature `((String) -> Unit)`, again representing that emitting a debug log message cannot be observed from within Q#.
```
Comment (alan-geller) [inline articles/libraries/standard/diagnostics.md]:
```suggestion
> `Message` has signature `(String -> Unit)`, again representing that emitting a debug log message cannot be observed from within Q#.
```

File: articles/libraries/standard/diagnostics.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+The <xref:microsoft.quantum.diagnostics.dumpmachine> and <xref:microsoft.quantum.diagnostics.dumpregister> callables instruct target machines to provide diagnostic information about all currently allocated qubits or specific register of qubits, respectively.
```
Comment (alan-geller) [inline articles/libraries/standard/diagnostics.md]:
```suggestion
The <xref:microsoft.quantum.diagnostics.dumpmachine> and <xref:microsoft.quantum.diagnostics.dumpregister> callables instruct target machines to provide diagnostic information about all currently allocated qubits or about a specific register of qubits, respectively.
```

File: articles/libraries/standard/index.md [hunk 6]
```diff
@@ -9,16 +9,10 @@ uid: microsoft.quantum.libraries.standard.intro
 ---
 # Introduction to the Q# Standard Libraries #
 
-Q# is supported by a range of different useful operations, functions, and user-defined types that comprise the Q# *standard library*.
-The Q# standard library is split into two main parts:
+Q# is supported by a range of different useful operations, functions, and user-defined types that comprise the Q# *standard libraries*.
+The [`Microsoft.Quantum.Development.Kit` NuGet package](https://www.nuget.org/packages/microsoft.quantum.development.kit) installed during [installation and validation](xref:microsoft.quantum.install) provides the Q# compiler, and parts of the standard library that are implemented by the target machine.
```
Comment (alan-geller) [inline articles/libraries/standard/index.md]:
```suggestion
The [`Microsoft.Quantum.Development.Kit` NuGet package](https://www.nuget.org/packages/microsoft.quantum.development.kit) installed during [installation and validation](xref:microsoft.quantum.install) provides the Q# compiler, and parts of the standard library that are implemented by the target machines.
```

File: articles/libraries/standard/licensing.md [hunk 7]
```diff
@@ -15,16 +15,20 @@ uid: microsoft.quantum.libraries.licensing
...
-
-There is also an extensive set of samples that explain and illustrate the use of functions and operations from the standard library and other libraries. These samples are licensed under the [MIT license](https://github.com/Microsoft/Quantum/blob/master/LICENSE.txt). The samples can be obtained from the [Microsoft/Quantum](https://github.com/Microsoft/Quantum) repository on GitHub.
+Microsoft's Quantum Development Kit also provides specialized library functions and operations that are licensed under a [Microsoft Research license](https://github.com/Microsoft/Quantum-NC/blob/master/LICENSE).
+These can be obtained from the [Microsoft/Quantum-NC](https://github.com/microsoft/quantum-nc) repository on GitHub.
 
+There is also an extensive set of samples that explain and illustrate the use of functions and operations from the standard library and other libraries.
```
Comment (alan-geller) [inline articles/libraries/standard/licensing.md]:
```suggestion
There are also many samples that explain and illustrate the use of functions and operations from the standard library and other libraries.
```

**Comment by cgranade** [comment]
Thanks for the review, @alan-geller!

**Pull request merged**